### PR TITLE
Es6

### PR DESCRIPTION
**Pull request by nicktate**
Converted part of legiond into es6 format. Need to finish converting the rest of the `lib/*` files

Made sure some of the callbacks followed the standard `(err, results...)` convention used in node
@normanjoyner

### CODE REVIEW

File: legiond.js [hunk 1]
```diff
@@ -1,165 +1,183 @@
...
+    }
 
-LegionD.prototype.join = function(event){
-    var self = this;
+    /*
+     * Retrieve all the known peers connected to the cluster. Strip
+     * pubkey & prime number before returning the node attributes
+     */
+    get_peers() {
+        return _.map(this.libraries.nodes.list, (node/*, name*/) => {
```
Comment (normanjoyner) [inline legiond.js]: remove commented out `name` param

File: lib/cache.js [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
+            clearTimeout(this.keys[key]);
+        }
+    }
 
-Cache.prototype.purge = function(){
-    _.each(this.keys, function(timer, key){
-        clearTimeout(timer);
-    }, this);
+    purge() {
+        _.each(this.keys, (timer/*, key*/) => {
```
Comment (normanjoyner) [inline lib/cache.js]: remove commented out `key` param

File: lib/network.js [hunk 3]
```diff
@@ -1,298 +1,305 @@
...
-    var clients = [];
+        const clients = [];
 
-    var unencryptable_events = [
-        'legiond.discovery',
-        'legiond.discovered',
-        'legiond.accept_node',
-        'legiond.node_accepted'
-    ]
+        const unencryptable_events = [
```
Comment (normanjoyner) [inline lib/network.js]: can we change this to `UNENCRYTABLE_EVENTS`

File: lib/cache.js (hunk 2 above)
Comment (nicktate) [inline lib/cache.js]: Leaving per our chat, it is clearer to see there could be a 2nd parameter even though we are not using it.

File: legiond.js (hunk 1 above)
Comment (nicktate) [inline legiond.js]: Leaving per our chat, it is clearer to see there could be a 2nd parameter even though we are not using it.

**Pull request merged**